File: CodingChallenges/CC_05_Space_invaders_py/Flower.py
# Daniel Shiffman - Translated  to Python by Ali Al Saqban
# http://codingtra.in
# http://patreon.com/codingtrain

# Space Invaders
# https://youtu.be/biN3v3ef-Y0

class Flower(object):
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.r = 30
        
        self.xdir = 1
    
    def grow(self):
        self.r = self.r + 2
    
    def shiftDown(self):
        self.xdir *= -1
        self.y += self.r
    
    def move(self):
        self.x = self.x + self.xdir
    
    def show(self):
        noStroke()
        fill(255,0,200,150)
        ellipse(self.x,self.y,self.r*2,self.r*2)
    

File: CodingChallenges/CC_005_Space_invaders/P5/flower.js
// Daniel Shiffman
// http://codingtra.in
// http://patreon.com/codingtrain
// Code for: https://youtu.be/biN3v3ef-Y0

function Flower(x, y) {
  this.x = x;
  this.y = y;
  this.r = 30;

  this.xdir = 1;

  this.grow = function() {
    this.r = this.r + 2;
  };

  this.shiftDown = function() {
    this.xdir *= -1;
    this.y += this.r;
  };

  this.move = function() {
    this.x = this.x + this.xdir;
  };

  this.show = function() {
    noStroke();
    fill(255, 0, 200, 150);
    ellipse(this.x, this.y, this.r * 2, this.r * 2);
  };
}
